### PR TITLE
CLI Cleanup 1

### PR DESCRIPTION
**Pull request by hydra**
- [x] Extract CLI settings table so that they can be used by other parts of the system (e.g. CMS)
- [x] Remove CLI backup `switch` code and variables.  This was always a pain because you actually had to remember to add any new PG to the CLI code rather than it being automatic.
- [x] Adjust and fix size/speed/normal optimizations.

SPRACINGF3NEO:
BEFORE
```
   text	   data	    bss	    dec	    hex	filename
 249606	   6580	  32992	 289178	  4699a	./obj/main/cleanflight_SPRACINGF3NEO.elf
```
AFTER
```
   text	   data	    bss	    dec	    hex	filename
 243952	   6584	  32988	 283524	  45384	./obj/main/cleanflight_SPRACINGF3NEO.elf
```

249606 - 243952 = 5654 bytes!

NAZE32:
BEFORE:
```region `FLASH' overflowed by 748 bytes```

AFTER:
```
   text	   data	    bss	    dec	    hex	filename
 127256	   1616	  18376	 147200	  23f00	./obj/main/cleanflight_NAZE.elf
```

126*1024 = 129024 + 748 = 129772 - 127256 = 2516 bytes!

### CODE REVIEW

**Comment by martinbudden** [comment]
I like the direction this is heading. Putting the copy in the PG declaration makes sense - I should have thought of that.

**Comment by hydra** [comment]
@martinbudden I updated the PR description to show the bytes saved by this PR.

File: src/main/fc/cli.c [hunk 1]
```diff
@@ -988,14 +187,24 @@ static void cliPrint(const char *str)
     bufWriterFlush(cliWriter);
 }
 
+static void cliPrintLine(const char *str)
+{
+    cliPrint(str);
+    cliPrint("\n\r");
```
Comment (mikeller) [inline src/main/fc/cli.c]: Could call `cliPrintBlankLine()` here.

File: src/main/fc/cli.c [hunk 2]
```diff
@@ -988,14 +187,24 @@ static void cliPrint(const char *str)
...
+}
+
+static void cliPrintBlankLine()
+{
+    cliPrint("\n\r");
```
Comment (mikeller) [inline src/main/fc/cli.c]: <pedantic>Should be `\r\n`</pedantic>

File: src/main/fc/settings.c [hunk 3]
```diff
@@ -0,0 +1,688 @@
+/*
```
Comment (mikeller) [inline src/main/fc/settings.c]: :+1 for extracting these

**Comment by hydra** [comment]
@mikeller Thanks for quick review, commits added with further flash savings, description updated.

File: src/main/fc/settings.c (hunk 3 above)
Comment (hydra) [inline src/main/fc/settings.c]: oddly, there was also actually a 32 byte saving in doing that when the ARRAYLEN usages were replaced with a reference to a static variable...

**Pull request merged**